### ISSUE TITLE
"Fast" bubble columns

### ISSUE THREAD

**Issue by Regnander** [sp614x/optifine#1895]
During Minecon Earth 2018, Bedrock developer Jason Major mentioned how the particles of bubble columns can be replaced with an animated texture in Bedrock Edition to improve performance. [(Watch it here.)](https://youtu.be/Jb4xSzTNd-I?t=10995) This sounds like something that would be perfect to implement in OptiFine due to the amounts of particles they create.

For reference, this is how the bubble columns look like in Bedrock Edition if you disable Fancy Graphics:

![bedrock-bubble-column](https://user-images.githubusercontent.com/17113053/46876192-fa273300-ce3d-11e8-810c-ace1ea7d30a8.gif)

**Comment by Madis0** [comment]
I wonder how well would it work with resource packs though? Would they need a separate texture for that?

Overall I like the idea though, would probably make it easier to debug for streams that are not continuously pushing ("flowing" and "static" water mixed).

**Comment by Regnander** [comment]
>I wonder how well would it work with resource packs though? Would they need a separate texture for that?

Yes, since the block model for the bubble column in Bedrock Edition consists of three textures:
- One that is wrapped around the column (the one with the larger and slower bubbles)
- One that is wrapped around the vertical pipe in the middle (the one with the smaller and faster bubbles)
- One for the top of the bubble column
